db/index.js

```diff
@@ -127,6 +127,38 @@ function create(options) {
     })
   }
 
+  var timerIdTtlScanner = setInterval(function() {
+    var currentUnixSeconds = Math.round(Date.now() / 1000)
+    function logError(err, result) {
+      if (err) console.error("@@@", err)
+    }
+    lazyStream(tableDb.createKeyStream({}), logError)
+        .join(function(tableNames) {
+          tableNames.forEach(function(name) {
+            getTable(name, false, function(err, table) {
+              if (err) return
+              if (!table.TimeToLiveDescription || table.TimeToLiveDescription.TimeToLiveStatus !== 'ENABLED') return
+
+              var itemDb = getItemDb(table.TableName)
+              var kvStream = lazyStream(itemDb.createReadStream({}), logError())
+              kvStream = kvStream.filter(function(item){
+                var ttl = item.value[table.TimeToLiveDescription.AttributeName]
+                return ttl && typeof ttl.N === 'string' && currentUnixSeconds > Number(ttl.N)
+              })
+              kvStream.join(function(items){
+                items.forEach(function(item) {
+                  itemDb.del(item.key)
+                })
+              })
+            })
+          })
+        })
+  }, 1000)
```

Comment:
I wonder how long the expiry job may run if there are many tables or large tables. 🤔 

I occasionally dump a lot of data into dynalite in order to exercise business logic code, in which case, table scan can easily take longer than 1s.

Btw., aws dynamodb expiry is not guaranteed to happen on the second, in fact it can be delayed by hours even a day.

Reply:
> Btw., aws dynamodb expiry is not guaranteed to happen on the second, in fact it can be delayed by hours even a day.

Yeah, that is true.
Currently, I set it `1000` to minimize test time.

> https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/howitworks-ttl.html
> TTL typically deletes expired items within 48 hours of expiration.

I am considering to make it configurable.

>  table scan can easily take longer than 1s.

I am also considering to optimize table scan.

Reply:
Per my comments below, I do still share overall concerns about efficiency – moreso seeing that folks like @dimaqq dump large amounts of data into this thing. I def think this can get figured out though!